core/src/test/java/org/apache/iceberg/TestManifestWriter.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg;
+
+import java.io.IOException;
+import java.util.UUID;
+import org.apache.iceberg.ManifestEntry.Status;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class TestManifestWriter extends TableTestBase {
```

Comment:
Technically, this doesn't have to extend `TableTestBase` but we need `writeManifest`. I think we can refactor `TableTestBase` a bit and introduce a separate parent class with the logic for writing/checking manifests/snapshots.

Reply:
+1 to extract out base classes for just manifest and snapshots.

Reply:
Either way is fine with me. I usually prefer to keep things simple.

Reply:
I wanted to refactor but it looks the changes won't be worth the effort after a closer look. For example, we will need to pass a partition spec and `FileIO` to write manifests that we can simply take from `table` in `TableTestBase`. Let's keep it as is for now.